Userland/Utilities/image.cpp

```diff
@@ -16,18 +16,37 @@
 #include <LibGfx/ImageFormats/QOIWriter.h>
 
 struct LoadedImage {
-    RefPtr<Gfx::Bitmap> bitmap;
+    Gfx::NaturalFrameFormat internal_format;
+    AK::Variant<RefPtr<Gfx::Bitmap>, RefPtr<Gfx::CMYKBitmap>> bitmap;
+
     Optional<ReadonlyBytes> icc_data;
 };
 
 static ErrorOr<LoadedImage> load_image(RefPtr<Gfx::ImageDecoder> const& decoder, int frame_index)
 {
-    return LoadedImage { TRY(decoder->frame(frame_index)).image, TRY(decoder->icc_data()) };
+    auto internal_format = decoder->natural_frame_format();
+
+    AK::Variant<RefPtr<Gfx::Bitmap>, RefPtr<Gfx::CMYKBitmap>> bitmap = TRY([&]() -> ErrorOr<decltype(bitmap)> {
```

Comment:
I kinda like this `decltype`, but at the same time I think that this is better (feel free to disagree):
`auto bitmap = []() -> ErrorOr<Variant<RefPtr<Gfx::Bitmap>, RefPtr<Gfx::CMYKBitmap>>> ...`

And again, no need for `AK`